### PR TITLE
[dev/mapping-tool] Gradle 7.0.2 and updated dependencies

### PR DESCRIPTION
**Pull request by Toffikk**
+ ported mapper build.gradle to build.gradle.kts

edit: and also fixed all deprecations in the mapper
edit 2: fixed all deprecations , and now the execution optimizations apply for paperclip task too! so building is faster

this was the last deprecation : 

```gradle
> Task :paperclip
Execution optimizations have been disabled for task ':paperclip' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/home/runner/work/Yatopia/Yatopia/Yatopia-Server/build/libs/yatopia-server-1.16.5-R0.1-SNAPSHOT.jar'. Reason: Task ':paperclip' uses this output of task ':yatopia-server:jar' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
Gradle detected a problem with the following location: '/home/runner/work/Yatopia/Yatopia/Yatopia-Server/build/libs/yatopia-server-1.16.5-R0.1-SNAPSHOT.jar'. Reason: Task ':paperclip' uses this output of task ':yatopia-server:jar' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem. This behaviour has been deprecated and is scheduled to be removed in Gradle 8.0. Execution optimizations are disabled to ensure correctness. See https://docs.gradle.org/7.0.2/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.
```
and it's now fixed!!!

### CODE REVIEW

File: mappings/mapper/settings.gradle.kts [hunk 1]
```diff
@@ -4,7 +4,7 @@
  * The settings file is used to specify which projects to include in your build.
  *
  * Detailed information about configuring a multi-project build in Gradle can be found
- * in the user manual at https://docs.gradle.org/6.5.1/userguide/multi_project_builds.html
+ * in the user manual at https://docs.gradle.org/7.0.0/userguide/multi_project_builds.html
  */
```
Comment (metabrixkt) [inline mappings/mapper/settings.gradle.kts]: new link returns 404, replace `7.0.0` with `7.0.2`

Comment (Toffikk) [reply mappings/mapper/settings.gradle.kts]: ok

Comment (Toffikk) [reply mappings/mapper/settings.gradle.kts]: done

**Review by josephworks — APPROVED** [review]

**Comment by Toffikk** [comment]
i will just do a small change and then it’s ready to merge 👍

**Pull request merged**